libraries/Microsoft.Bot.Builder.Dialogs/Prompts/OAuthPrompt.cs

```diff
@@ -336,18 +337,18 @@ private async Task SendOAuthCardAsync(ITurnContext turnContext, IMessageActivity
             }
             else if (!prompt.Attachments.Any(a => a.Content is OAuthCard))
             {
-                var cardActionType = ActionTypes.Signin;
                 string signInLink = null;
 
-                if (turnContext.Activity.IsFromStreamingConnection())
+                var activity = turnContext.Activity;
+                if (activity.ChannelId == Channels.Emulator)
                 {
-                    signInLink = await adapter.GetOauthSignInLinkAsync(turnContext, _settings.OAuthAppCredentials, _settings.ConnectionName, cancellationToken).ConfigureAwait(false);
+                    var channelData = JsonConvert.DeserializeObject<Dictionary<string, string>>(activity.ChannelData.ToString());

```

Comment:
>ToString [](start = 117, length = 8)

this is highly unlikely to be correct.  ChannelData is an object, and should be deserialized as a JObject...